modules/ssl/mod_ssl.c

```diff
@@ -481,18 +481,23 @@ static SSLConnRec *ssl_init_connection_ctx(conn_rec *c,
      * includes mod_proxy's later run_pre_connection call), sslconn->dc should
      * be preserved if it's already set.
      */
+    /* icing: this method primarily initializes "sslconn" for a connection
+     * if it is not already there. But calling it with "new_proxy" re-inits
+     * the sslconn's dirconf parts. So "reinit" would be a more apt name.
+     */
     if (!sslconn) {
         sslconn = apr_pcalloc(c->pool, sizeof(*sslconn));
         need_setup = 1;
     }
-    else if (!new_proxy) {
+    else if (!reinit) {
         return sslconn;
     }
 
     /* Reinit dc in any case because it may be r->per_dir_config scoped
      * and thus a caller like mod_proxy needs to update it per request.
      */
     if (per_dir_config) {
+        ap_assert(c->outgoing);
```

Comment:
Isn't upgrade_connection()->ssl_init_ssl_connection()->ssl_init_connection_ctx() a call chain where we'd fall through here with r->per_dir_config?
I don't the hard requirement on c->outgoing being set here, why enfore/assert it?

Reply:
I think you are correct.

Reply:
You are right. There is the incoming upgrade situation as well. Quite some call chains going on.

Will remove the assert.